stdlib/influxdata/influxdb/cardinality_test.go

```diff
@@ -0,0 +1,372 @@
+						IsRelative: true,
+					},
+					Now: now,
+				},
+			},
+			want: testutil.Want{
+				Params: url.Values{
+					"org": []string{"influxdata"},
+				},
+				Query: `package main
+
+
+import influxdb "influxdata/influxdb"
+
+influxdb.cardinality(bucket: "telegraf", start: 2020-10-22T09:29:00Z, stop: 2020-10-22T09:30:00Z)`,
+				Tables: defaultTablesFn,
+			},
+		},
+		{
+			name: "basic query with org id and bucket id",
+			spec: &influxdb.CardinalityProcedureSpec{
+				Config: influxdb.Config{
+					Org:    influxdb.NameOrID{ID: "97aa81cc0e247dc4"},
+					Bucket: influxdb.NameOrID{ID: "1e01ac57da723035"},
+					Token:  "mytoken",
+				},
+				Bounds: flux.Bounds{
+					Start: flux.Time{
+						IsRelative: true,
+						Relative:   -time.Minute,
+					},
+					Stop: flux.Time{
+						IsRelative: true,
+					},
+					Now: now,
+				},
+			},
+			want: testutil.Want{
+				Params: url.Values{
+					"orgID": []string{"97aa81cc0e247dc4"},
+				},
+				Query: `package main
+
+
+import influxdb "influxdata/influxdb"
+
+influxdb.cardinality(bucketID: "1e01ac57da723035", start: 2020-10-22T09:29:00Z, stop: 2020-10-22T09:30:00Z)`,
+				Tables: defaultTablesFn,
+			},
+		},
+		{
+			name: "basic query with absolute time range",
+			spec: &influxdb.CardinalityProcedureSpec{
+				Config: influxdb.Config{
+					Org:    influxdb.NameOrID{Name: "influxdata"},
+					Bucket: influxdb.NameOrID{Name: "telegraf"},
+					Token:  "mytoken",
+				},
+				Bounds: flux.Bounds{
+					Start: flux.Time{
+						Absolute: mustParseTime("2018-05-30T09:00:00Z"),
+					},
+					Stop: flux.Time{
+						Absolute: mustParseTime("2018-05-30T10:00:00Z"),
+					},
+					Now: now,
+				},
+			},
+			want: testutil.Want{
+				Params: url.Values{
+					"org": []string{"influxdata"},
+				},
+				Query: `package main
+
+
+import influxdb "influxdata/influxdb"
+
+influxdb.cardinality(bucket: "telegraf", start: 2018-05-30T09:00:00Z, stop: 2018-05-30T10:00:00Z)`,
+				Tables: defaultTablesFn,
+			},
+		},
+		{
+			name: "filter query",
+			spec: &influxdb.CardinalityProcedureSpec{
+				Config: influxdb.Config{
+					Org:    influxdb.NameOrID{Name: "influxdata"},
+					Bucket: influxdb.NameOrID{Name: "telegraf"},
+					Token:  "mytoken",
+				},
+				Bounds: flux.Bounds{
+					Start: flux.Time{
+						IsRelative: true,
+						Relative:   -time.Minute,
+					},
+					Stop: flux.Time{
+						IsRelative: true,
+					},
+					Now: now,
+				},
+				PredicateSet: influxdb.PredicateSet{{
+					ResolvedFunction: interpreter.ResolvedFunction{
+						Fn:    executetest.FunctionExpression(t, `(r) => r._value >= 0.0`),
+						Scope: valuestest.Scope(),
+					},
+				}},
+			},
+			want: testutil.Want{
+				Params: url.Values{
+					"org": []string{"influxdata"},
+				},
+				Query: `package main
+
+
+import influxdb "influxdata/influxdb"
+
+influxdb.cardinality(
+	bucket: "telegraf",
+	start: 2020-10-22T09:29:00Z,
+	stop: 2020-10-22T09:30:00Z,
+	predicate: (r) => {
+		return r["_value"] >= 0.0
```

Comment:
Nit: could you change this to `r._value`. I know `r["_value"]` is valid flux but I'd like us to use the former throughout the codebase when possible.

Reply:
I mostly did this because it's automated code that generates this and I didn't know what the conditions were to check for a valid identifier. I didn't want to accidentally generate invalid flux if there was a space or something else.